aten/src/ATen/core/SymbolicIntNode.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <c10/macros/Macros.h>
+#include <c10/util/Exception.h>
+#include <vector>
+#include <ATen/core/SymInt.h>
+#include <mutex>
+
+namespace c10 {
+
+class TORCH_API SymbolicIntNode {
+    public:
+        c10::SymInt toSymInt();
+        static bool is_symbolic(int64_t data) {
+            return SYM_TAG_MASK & data;
+        }
+
+        const static int64_t SYM_TAG_MASK = 1LL << 63;
+
+        virtual ~SymbolicIntNode() {};
+    private:
+        int64_t data_;
```

Comment:
I am confused. SymInt is the thing that fits in `int64_t` (and is where all the bitmasking logic ought to live). SymbolicIntNode is supposed to be the boxed thing that you can put all of the random extra IR bits you need to identify the object in question!